### PR TITLE
Adds customer specs and other useful additions

### PR DESCRIPTION
**Pull request by mcrisologo**
This PR adds the ability to create a customer using the Square API.

In addition to this, we added also some other useful things:

- Add Square as available payment method for Solidus
- Add vcr, webmock and pry to development dependencies
- Update rubocop config

### CODE REVIEW

**Comment by mcrisologo** [comment]
Note: 
Fetching initial recording is a bit problematic. 
Seems like pinging to the sandbox API is intermittent (even on console).
Using stubs instead.

File: spec/support/vcr.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+require 'vcr'
+require 'webmock'
+
+VCR.configure do |config|
+  config.ignore_localhost = true
+  config.cassette_library_dir = "spec/fixtures/vcr_cassettes"
+  config.hook_into :webmock
+  config.configure_rspec_metadata!
+  config.allow_http_connections_when_no_cassette = false
```
Comment (DanielePalombo) [inline spec/support/vcr.rb:11]: [This part](https://github.com/solidusio-contrib/solidus_paypal_marketplace/blob/main/spec/support/vcr.rb#L58-L59) would be useful to invalidate the cassettes.

File: spec/fixtures/vcr_cassettes/customers/create/valid.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        encoding: UTF-8
+        string: '{"given_name":"John","family_name":"Doe","email_address":"email1@example.com","address":{"address_line_1":"A
+        Different Road","address_line_2":"A Different Road","locality":"Herndon","postal_code":"10002","country":"US"},"phone_number":"555-555-0199","reference_id":"1"}'
+      headers:
+        Accept:
+          - application/json
+        Content-Type:
+          - application/json; charset=utf-8
+        Authorization:
+          - Bearer EAAAEJ2wQT1qUQIvMGlImkhphBakgTk-r8rhyznGQ_5IvSKFpKk4vmMexZRGAIgR
```
Comment (DanielePalombo) [inline spec/fixtures/vcr_cassettes/customers/create/valid.yml:16]: Please filter out the sensitive data with this [code](https://github.com/solidusio-contrib/solidus_paypal_marketplace/blob/main/spec/support/vcr.rb#L15-L17)

File: spec/services/solidus_square/customers/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  subject(:service) do
+    described_class.call({ client: '', spree_user: spree_user, spree_address: spree_address })
+  end
+
+  let(:spree_user) { create(:user_with_addresses) }
+  let(:spree_address) { spree_user.addresses.first }
+
+  context 'when customer already exists' do
+    before do
+      allow_any_instance_of(described_class).to receive(:search_customer).and_return({ 'id' => 111 })
```
Comment (DanielePalombo) [inline spec/services/solidus_square/customers/create_spec.rb:15]: Do you mind replacing these `allow_any_instance_of``s in favor of Double and specific allow?

File: spec/support/solidus.rb [hunk 4]
```diff
@@ -0,0 +1 @@
+#require 'spree/testing_support/order_walkthrough'
```
Comment (DanielePalombo) [inline spec/support/solidus.rb:1]: This file can be removed

File: app/models/solidus_square/gateway.rb [hunk 5]
```diff
@@ -13,5 +13,9 @@ def initialize(options)
         environment: options[:environment]
       )
     end
+
+    def create_customer(user, address)
```
Comment (DanielePalombo) [inline app/models/solidus_square/gateway.rb:18]: missing test

**Review by DanielePalombo — CHANGES_REQUESTED** [review]
There are some requested changes.

File: app/models/solidus_square/payment_method.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module SolidusSquare
+  class PaymentMethod < SolidusSupport.payment_method_parent_class
```
Comment (DanielePalombo) [inline app/models/solidus_square/payment_method.rb:4]: This is also related to #7

**Pull request merged**